src/Stateless/StateMachine.cs

```diff
@@ -5,6 +5,16 @@
 
 namespace Stateless
 {
+    /// <summary>
+    /// Enum for the different modes used when Fire-ing a trigger
+    /// </summary>
+    public enum FiringMode
+    {
+        /// <summary> Use immediate mode when the queing of trigger events are not needed. Care must be taken when using this mode, as there is no run-to-completion guaranteed.</summary>
+        Immediate,
+        /// <summary> Use the queued Fire-ing mode when run-to-completion is required. This is the recommended mode.</summary>
+        Queued
```

Comment:
Although a queue is used for the implementation, would it be clearer to call this something like `OnCompletedTransition`?

Reply:
:+1: